### PR TITLE
Sequence points and symbolicate improvements

### PR DESCRIPTION
**Pull request by esdrubal**
Renamed symbolicate tool to mono-symbolicate.
Sequence points on release AOT are now stored entirely on a separate .msym file.
.msym files avoid a data overhead on the packages that are deployed.
.msym files can be used by symbolicate tool to add filename and line to stacktraces.
Moved some mini/seq-points.c/.h code to metadata/seq-points-data.c/.h.
Created an icall that is be called by the symbolicate tool to obtain an IL offset from method tokens and native offset.
Updated symbolicate tool to use .msym files when stack trace has method tokens and native offsets.
Fixed a crash when calling the tool with directories.
Wrote man page and install script for the symbolicate tool.
Added --aot option gen-seq-points-file.

### CODE REVIEW

File: man/mono-symbolicate.1 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+.\" 
+.\" mono-symbolicate manual page.
+.\" Copyright 2006 Novell
```
Comment (jbevain) [inline man/mono-symbolicate.1]: hrm :)

Comment (esdrubal) [inline man/mono-symbolicate.1]: Fixed, thanks!

File: scripts/mono-symbolicate [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec /usr/local/bin/mono $MONO_OPTIONS /usr/local/lib/mono/4.5/mono-symbolicate.exe "$@"
```
Comment (tritao) [inline scripts/mono-symbolicate]: This should be a `.in` and use the macros to have the correct path when it's installed. See https://github.com/esdrubal/mono/blob/aotsp/scripts/mcs.in for example.

**Comment by vargaz** [comment]
Some comments;
- I think it would be better to add an option to the aot compiler to turn this on, since its not very useful on desktop.
- The changes which modify each other should be squashed together

**Comment by esdrubal** [comment]
@vargaz Fixed changes that change each other on aot-compiler.c and aot-runtime.c.
Added MONO_DEBUG option gen-seq-points-aot-file.

**Comment by vargaz** [comment]
I meant an --aot= option, MONO_DEBUG options are for debugging the runtime etc.

**Comment by esdrubal** [comment]
Changed it to an --aot option.

File: mono/metadata/icall.c [hunk 3]
```diff
@@ -7677,6 +7678,16 @@ ves_icall_System_ComponentModel_Win32Exception_W32ErrorMessage (guint32 code)
 	return message;
 }
 
+ICALL_EXPORT int
+ves_icall_System_StackFrame_GetILOffsetFromFile (MonoString *path, int methodToken, int nativeOffset)
+{
+	guint32 il_offset;
+	if (seq_point_data_get_il_offset (mono_string_to_utf8 (path), methodToken, nativeOffset, &il_offset))
```
Comment (kumpera) [inline mono/metadata/icall.c]: This leaks memory.

**Comment by kumpera** [comment]
build

**Pull request merged**